test/index.ts

```diff
@@ -189,6 +192,127 @@ describe('Storage', () => {
+      serviceAccountEmail: SERVICE_ACCOUNT_EMAIL,
+      state: 'ACTIVE',
+      timeCreated: '20190101T00:00:00Z',
+      updated: '20190101T00:00:00Z',
+    };
+    const response = {
+      secret: 'my-secret',
+      metadata: metadataResponse,
+    };
+
+    it('should make correct API request', done => {
+      storage.request = (
+        reqOpts: DecorateRequestOptions,
+        callback: Function
+      ) => {
+        assert.strictEqual(reqOpts.method, 'POST');
+        assert.strictEqual(
+          reqOpts.uri,
+          `/projects/${storage.projectId}/hmacKeys`
+        );
+        assert.strictEqual(
+          reqOpts.qs.serviceAccountEmail,
+          SERVICE_ACCOUNT_EMAIL
+        );
+
+        callback(null, response);
+      };
+
+      storage.createHmacKey(SERVICE_ACCOUNT_EMAIL, done);
+    });
+
+    it('should throw without a serviceAccountEmail', () => {
+      assert.throws(
+        () => storage.createHmacKey(),
+        /must be a service account email/
+      );
+    });
+
+    it('should throw when first argument is not a string', () => {
+      assert.throws(
+        () =>
+          storage.createHmacKey({
+            userProject: 'my-project',
+          }),
+        /must be a service account email/
+      );
+    });
+
+    it('should honor the userProject option', async () => {
+      const options = {
+        userProject: 'my-project',
+      };
+
+      storage.request = sinon
+        .stub()
+        .returns((_reqOpts: {}, callback: Function) => callback());
+
+      await storage.createHmacKey(SERVICE_ACCOUNT_EMAIL, options);
+      const reqArg = storage.request.firstCall.args[0];
+      assert.strictEqual(reqArg.qs.userProject, options.userProject);
+    });
+
+    it('should invoke callback with a secret and an HmacKey instance', done => {
+      storage.request = (_reqOpts: {}, callback: Function) => {
+        callback(null, response);
+      };
+
+      storage.createHmacKey(
+        SERVICE_ACCOUNT_EMAIL,
+        (err: Error, hmacKey: HmacKey, secret: string) => {
+          assert.ifError(err);
+          assert.strictEqual(secret, response.secret);
+          assert(hmacKey instanceof HmacKey);
+          assert.strictEqual(hmacKey.metadata, metadataResponse);
+          assert.strictEqual(hmacKey.accessId, metadataResponse.accessId);
+          done();
+        }
+      );
+    });
+
+    it('should invoke callback with raw apiResponse', done => {
+      storage.request = (_reqOpts: {}, callback: Function) => {
+        callback(null, response);
+      };
+
+      storage.createHmacKey(
+        SERVICE_ACCOUNT_EMAIL,
+        (
+          err: Error,
+          _hmacKey: HmacKey,
+          _secret: string,
+          apiResponse: HmacKeyResourceResponse
+        ) => {
+          assert.ifError(err);
+          assert.strictEqual(apiResponse, response);
+          done();
+        }
+      );
+    });
+
+    it('should execute callback with request error', done => {
```

Comment:
Please check that it also gets the apiResponse argument as well.

Reply:
fixed!